addons/account/models/account_move.py

```diff
@@ -2052,6 +2052,17 @@ def _check_journal_type(self):
             if record.is_sale_document() and journal_type != 'sale' or record.is_purchase_document() and journal_type != 'purchase':
                 raise ValidationError(_("The chosen journal has a type that is not compatible with your invoice type. Sales operations should go to 'sale' journals, and purchase operations to 'purchase' ones."))
 
+    @api.constrains('commercial_partner_id', 'line_ids')
+    def _check_partner_id(self):
+        for move in self:
+            if move.is_invoice(include_receipts=True):
+                if any(
+                    move.commercial_partner_id != line.partner_id.commercial_partner_id
```

Comment:
I understand why I need to look at the commercial partner for the move, but not for the line

Reply:
I agree, but then this "test" doesn't agree https://runbot.odoo.com/runbot/batch/725714/build/13472764 and I'm not sure what is wrong with it.
So it creates an invoice with Partner A, and all the lines set to Partner A in the create vals.
Then, the partner is changed somewhere to take the commercial partner on the move but not on the lines.

Reply:
@smetl should be good now, I don't know why I didn't see the reason of the issue earlier...